File: src/ocr/catalog.ts
import { MODEL_CATALOG, type ModelProvider } from '../config/model-catalog';

export function flattenCatalogModels(): Array<{ provider: ModelProvider; modelId: string; label: string }> {
  const models: Array<{ provider: ModelProvider; modelId: string; label: string }> = [];
  (Object.keys(MODEL_CATALOG) as ModelProvider[]).forEach((provider) => {
    MODEL_CATALOG[provider].forEach((model) => {
      models.push({ provider, modelId: model.id, label: model.label });
    });
  });
  return models;
}

File: src/ocr_mini_bench/ocr/catalog.py
"""Flatten the provider/model catalog. Mirrors `src/ocr/catalog.ts`."""

from __future__ import annotations

from dataclasses import dataclass

from ..config.model_catalog import MODEL_CATALOG, ModelProvider


@dataclass(frozen=True)
class FlatCatalogModel:
    provider: ModelProvider
    model_id: str
    label: str


def flatten_catalog_models() -> list[FlatCatalogModel]:
    out: list[FlatCatalogModel] = []
    for provider, models in MODEL_CATALOG.items():
        for option in models:
            out.append(FlatCatalogModel(provider=provider, model_id=option.id, label=option.label))
    return out
